libs/wire-api/src/Wire/API/Call/TURN.hs

```diff
@@ -109,13 +112,19 @@ modelRtcConfiguration = Doc.defineModel "RTCConfiguration" $ do
     Doc.description "Number of seconds after which the configuration should be refreshed (advisory)"
 
 instance ToJSON RTCConfiguration where
-  toEncoding (RTCConfiguration srvs ttl) =
-    pairs ("ice_servers" .= srvs <> "ttl" .= ttl)
+  toJSON (RTCConfiguration srvs ttl) =
```

Comment:
There was a potential bug waiting here: The [documentation of ToJSON](https://www.stackage.org/haddock/lts-15.13/aeson-1.4.7.1/Data-Aeson.html#t:ToJSON) says:

> Instances in general must specify `toJSON` and should (but don't need to) specify `toEncoding`.

This is annoyingly not enforced by the compiler and leads to a broken `toJSON` implementation that violates the roundtrip property.

In prod, everything works because the handlers call `toEncoding` directly, but an innocent refactoring could break this.

Reply:
So...  Write more QC tests for all instances whether `toEncoding` and `toJSON` are in sync?  (That sounds like a bit of a waste...)

Reply:
Normal roundtrip tests guarantee that `toJSON` is implemented correctly and the default implementation of `toEncoding` uses that. But yeah, if both are implemented manually, we don't check whether they are in sync.

We could, but I'm not sure it's worth it. Then we should probably just not implement `toEncoding`. I don't think the performance benefit is significant enough to justify the additional code (and risk). Some day, the instances will be derived generically anyways, with a correct and efficient `toEncoding`.

Reply:
The only place where we still use `toEncoding` is the instance for `List1`, and it seems to be correct there.

Reply:
This looks like one of those conversations that we'll come back to after something has gone wrong, and think: "yes, we should have acted on that thought."  :)

But I'm not sure how.  Write an hlint rule that disallows manual implementations of `toEncoding`?

Reply:
For that we'd need to run hlint first of all. I don't think there is a quick and easy solution to this.

Reply:
I'll also send a short announcement that `toEncoding` does not make `toJSON`  unnecessary.